client/src/infographic/TableOfContents.js

```diff
@@ -2,105 +2,68 @@ import _ from "lodash";
 import React from "react";
 
 import {
-  Details,
+  StatelessDetails,
   create_text_maker_component,
   UnlabeledTombstone,
+  LinkStyled,
 } from "src/components/index.js";
 
 import { separatorColor } from "src/core/color_defs.js";
 
-import { infograph_options_href_template } from "./infographic_link.js";
-
 import text from "./TableOfContents.yaml";
 
 const { TM } = create_text_maker_component(text);
 
 export default class TableOfContents extends React.Component {
-  state = {
-    initial_open: false,
-  };
-
-  handle_click = () => {
-    this.setState((prev_state) => ({
-      initial_open: !prev_state.initial_open,
-    }));
-  };
-
+  constructor(props) {
+    super(props);
+    this.state = {
+      is_open: false,
+    };
+  }
```

Comment:
Just wondering why we prefer to do it like this.

(I'd assume it's because of consistency, and it allows us to add props into the initial state though?)

Reply:
Pretty much. Consistency, and the minimal number of line changes needed if props need to factor in to the initial state in the future.

Same idea as adding trailing commas to array/object definitions (at least multi-line ones). They're optional, but the fiddly little reason for linters/formatter rules to always enforce them is so that a future commit adding a new item to the array will only touch the new line and won't have to touch an old line just to add a new comma. This is most specifically for the sake of diffs and git blames, cutting down on the noise and keeping it just to the substantial change.

Reply:
Makes sense, thanks 👍